# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Recover.cpp
# Commit: 11fc8f557a2a
# Full Hash: 11fc8f557a2a7cf3c89e7c1396316aabd9725778
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 9bfa6023a553 -r 11fc8f557a2a js/src/jit/Recover.cpp
--- a/js/src/jit/Recover.cpp	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/jit/Recover.cpp	Sat Mar 06 01:05:21 2021 +0000
@@ -1571,9 +1571,9 @@
 bool RNewArray::recover(JSContext* cx, SnapshotIterator& iter) const {
   RootedObject templateObject(cx, &iter.read().toObject());
   RootedValue result(cx);
-  RootedObjectGroup group(cx, templateObject->group());
+  RootedShape shape(cx, templateObject->shape());
 
-  ArrayObject* resultObject = NewArrayWithGroup(cx, count_, group);
+  ArrayObject* resultObject = NewArrayWithShape(cx, count_, shape);
   if (!resultObject) {
     return false;
   }